kotlinx-coroutines-core/common/src/flow/terminal/Collect.kt

```diff
@@ -30,6 +30,30 @@ import kotlin.jvm.*
 @ExperimentalCoroutinesApi // tentatively stable in 1.3.0
 public suspend fun Flow<*>.collect() = collect(NopCollector)
 
+/**
+ * Terminal flow operator that [launches][launch] the [collection][collect] of the given flow in the [scope].
+ * It is a shorthand for `scope.launch { flow.collect() }`.
+ *
+ * This operator is usually used with [onEach], [onCompletion] and [catch] operators to process all emitted values
+ * handle an exception that might occur in the upstream flow or during processing, for example:
+ * ```
+ * flow
+ *     .onEach { value -> updateUi(value) }
+ *     .onCompletion { cause -> if (cause == null) updateUi("Done") }
```

Comment:
Consider the following `onCompletion` to highlight the use-case of always updating ui on any kind of completion and to remove duplicated `updateUi` call from `catch`:
```
.onCompletion { cause -> updateUi(if (cause == null) "Done" else "Fail") }
```